### PR TITLE
[AD-1043] [JDBC] Bug: Build not working in manual workflow action.

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[JDBC] Bug: Build not working in manual workflow action.

### Description

Mark SSH client tests to only run when integration testing enabled.

### Related Issue

https://bitquill.atlassian.net/browse/AD-1043

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -122,6 +122,12 @@ jobs:
       with:
         name: output
         path: output
+    - name: Upload Raw Test Results
```
Comment (birschick-bq) [inline .github/workflows/gradle.yml:125]: Add this to help debug in the future when the test result details are not usable.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -141,6 +147,7 @@ jobs:
         generate-branches-badge: true
     - name: Commit Coverage Badge
       uses: stefanzweifel/git-auto-commit-action@v4
+      if: ${{ ! github.event.inputs.mavenFilesPreparation }}
```
Comment (birschick-bq) [inline .github/workflows/gradle.yml:150]: Disables the commit of the code coverage badge when running the Maven on-demand workflow.

File: src/test/java/software/amazon/documentdb/jdbc/sshtunnel/DocumentDbSshTunnelClientTest.java [hunk 3]
```diff
@@ -57,6 +57,7 @@ void testConstructorDestructor() throws Exception {
     }
 
     @Test
+    @Tag("remote-integration")
```
Comment (birschick-bq) [inline src/test/java/software/amazon/documentdb/jdbc/sshtunnel/DocumentDbSshTunnelClientTest.java:60]: Ensure to only run this test when integration is enabled.

**Review by alinaliBQ — APPROVED** [review]
lgtm

**Pull request merged**